src/api.ts

```diff
@@ -94,4 +99,57 @@ export const getOpenCommitLatestVersion = async (): Promise<
   }
 };
 
+function generatePrefix(): string | undefined {
+  const prefix = config?.prefix
+
+  if (prefix === undefined) {
+    return undefined;
+  }
+
+  const prefixIsRegexString = prefix.startsWith('/') && prefix.endsWith('/');
+
+  if (prefixIsRegexString) {
+    try {
+      return generatePrefixFromRegex(prefix);
+    } catch (error) {
+      console.error(`Failed to generate prefix from regex: ${error}`);
+      return undefined;
+    }
+  }
+
+  return prefix;
+}
+
 export const api = new OpenAi();
+
+
+function generatePrefixFromRegex(regex: string): string | undefined {
+
+  // We currently only support regex input from git branch name
+
+  const branch = getCurrentGitBranch();
+
+  if (branch === undefined) {
+    return undefined;
+  }
+
+  const regexWithoutSlashes = regex.slice(1, -1);
+  const regexObject = new RegExp(regexWithoutSlashes);
+  const match = branch.match(regexObject);
+
+  if (match === null) {
+    return undefined;
+  }
+
+  return match[0];
```

Comment:
The example regex had () such as /^([a-zA-Z]+-\d+)/. Is it expecting the whole match if the parentheses are missing, or the first parentheses if the parentheses are present?
```suggestion
  r return match.length > 1 ? match[1] : match[0];
```